### ISSUE TITLE
Stack trace doesn't alway match code

### ISSUE THREAD

**Issue by vsmenon** [dart-lang/webdev#1250]
![Screen Shot 2021-02-15 at 6 55 41 PM](https://user-images.githubusercontent.com/2119553/108014663-3ac18900-6fc3-11eb-9176-556972ffa8bc.png)

Note that the method on the call stack doesn't match the code in the window.  I'm seeing the mismatch up and down the stack in this case.

This is with:
```
> flutter --version
Flutter 1.26.0-17.5.pre • channel beta • https://github.com/flutter/flutter.git
Framework • revision 1fe38dcb5f (5 days ago) • 2021-02-10 16:25:47 -0800
Engine • revision d4453f6018
Tools • Dart 2.12.0 (build 2.12.0-259.9.beta)
```

running the gallery sample ( https://github.com/flutter/gallery ).

**Comment by grouma** [comment]
Do you have repro steps? 

I set a breakpoint in the same location and hit it a bunch of times but I"m not seeing the same behavior:
<img width="1420" alt="Screen Shot 2021-02-16 at 5 56 11 PM" src="https://user-images.githubusercontent.com/1840773/108145597-900fa000-7080-11eb-9c42-8fede28f5cb8.png">


I'm currently testing with:
```
Flutter 1.27.0-2.0.pre.124 • channel master • git@github.com:flutter/flutter.git
Framework • revision a2a77285dc (66 minutes ago) • 2021-02-16 19:51:02 -0500
Engine • revision 9bcb3bfb0e
Tools • Dart 2.13.0 (build 2.13.0-30.0.dev)
```

Is it possible you have an old version of VS Code? @DanTup any thoughts?

**Comment by DanTup** [comment]
I don't think much has changed in this area (for VS Code or the extension) anytime recently, so suspect it's not related to the versions of those.

I also wasn't able to repro, although it's not clear how @vsmenon paused - it says "paused on pause" in the screenshot suggesting the pause button may have been clicked? Although doing so I couldn't get it to pause in the same place. If I add a breakpoint in the top frame shown in that screenshot, everything lines up for me (although it's a different call stack to get to that performRebuild method).

If you have repro steps, I can try again. Otherwise you could try to capture a log using the **Dart: Capture Debugging Logs** command in VS Code. This will buffer all of the traffic over the VM service and then once you click **Cancel** on the logging notification, open that in a new file.

**Comment by grouma** [comment]
>  it says "paused on pause" in the screenshot suggesting the pause button may have been clicked?

Oh interesting.

@DanTup what is the default behavior when pausing? I assume we jump to the source file referenced in the top frame.

I'm wondering if @vsmenon just paused the application and we didn't jump to the right source file or he happened to open up a different source file. This could be a non-issue?

**Comment by vsmenon** [comment]
I'll try again when I get a chance.  I stopped at a breakpoint in `main.dart` and stepped into either `runApp` or `build` and just continued stepping.  At some point, I noticed I was stopped on a comment and noticed the trace looked entirely off.

**Comment by DanTup** [comment]
> @DanTup what is the default behavior when pausing? I assume we jump to the source file referenced in the top frame.

Yep, though not directly. When clicking pause, the only thing that happens specifically is a pause request is sent to the VM. The rest of the behaviour all comes from the VM later sending a pause event (which is just the same as when it hits a breakpoint, but with a different type). In turn that's sent back to VS Code, which then triggers it asking us for a stack trace (which we satisfy the first frame from the topFrame in the last pause event, then the rest by calling getStack()).

**Comment by DanTup** [comment]
Though based on the comment above, it might be that stepping is reported as "paused on pause". If so, that's a bug (assuming the VM distinguishes between them), as we should report it back to VS Code as a "step". I'll check this tomorrow.
